weldx/core.py

```diff
@@ -822,22 +822,64 @@ class GenericSeries:
     """Describes a quantity depending on one or more parameters."""
 
     _allowed_variables: list[str] = []
-    """Allowed variable names"""
+    """A list of allowed variable names.
+
+    The names are only relevant if the created `GenericSeries` is based on expressions.
+    If the expression contains any other variable name that is not part of the list an,
+    an exception is raised. It is not required that an expression includes all these
+    variables. Additionally, the expression can contain other symbols if they are used
+    as parameters.
+    """
     _required_variables: list[str] = []
-    """Required variable names"""
+    """A list of required variable names.
+
+    The names are only relevant if the created `GenericSeries` is based on expressions.
+    If one or more variables are missing in the expression, an exceptions is raised.
+    Note, that the required symbols must be variables of the expression. Using one or
+    more as a parameter will also trigger an exception.
+    """
 
     _evaluation_preprocessor: dict[str, Callable] = {}
-    """Function that should be used to adjust a var. input - (f.e. convert to Time)"""
+    """A dictionary of preprocessor functions for input variables during evaluation.
+
+    When evaluating a `GenericSeries`, the passed keyword arguments are checked against
+    the dictionaries keys. If a match is found, the corresponding preprocessor function
+    is called with the variables value and returns the updated value. As an example,
+    this can be used to support multiple time formats. The key might be ``t`` and the
+    preprocessor function would turn the original time data into an equivalent
+    `xarray.DataArray`.
+    """
```

Comment:
```suggestion
    """A mapping of variable names to preprocessor functions are applied prior to evaluation.

    When calling a `GenericSeries.evaluate`, the passed keyword arguments are checked against
    the dictionaries keys. If a match is found, the corresponding preprocessor function
    is called with the variables value and returns the updated value. As an example,
    this can be used to support multiple time formats. The key might be ``t`` and the
    preprocessor function would turn the original time data into an equivalent
    `xarray.DataArray`.
    """
```